services/121-service/test/payment/do-payment-safaricom.test.ts

```diff
@@ -0,0 +1,247 @@
+    Male60: 0,
+    Female60: 0,
+    maleTotal: 0,
+    femaleTotal: 0,
+    householdMembersDisability: 'no',
+    disabilityAmount: 0,
+    householdMembersChronicIllness: 'no',
+    chronicIllnessAmount: 0,
+    householdMembersPregnantLactating: 'no',
+    pregnantLactatingAmount: 0,
+    habitableRooms: 0,
+    tenureStatusOfDwelling: 'Owner occupied',
+    ownerOccupiedState: 'purchased',
+    ownerOccupiedStateOther: 0,
+    rentedFrom: 'individual',
+    rentedFromOther: 0,
+    constructionMaterialRoof: 'tin',
+    ifRoofOtherSpecify: 31213,
+    constructionMaterialWall: 'tiles',
+    ifWallOtherSpecify: 231312,
+    constructionMaterialFloor: 'cement',
+    ifFloorOtherSpecify: 'asdsd',
+    dwellingRisk: 'fire',
+    ifRiskOtherSpecify: 123213,
+    mainSourceOfWater: 'lake',
+    ifWaterOtherSpecify: 'dasdas',
+    pigs: 'no',
+    ifYesPigs: 123123,
+    chicken: 'no',
+    mainModeHumanWasteDisposal: 'septic_tank',
+    ifHumanWasteOtherSpecify: 31213,
+    cookingFuel: 'electricity',
+    ifFuelOtherSpecify: 'asdsda',
+    Lighting: 'electricity',
+    ifLightingOtherSpecify: 'dasasd',
+    householdItems: 'none',
+    excoticCattle: 'no',
+    ifYesExoticCattle: 12231123,
+    IndigenousCattle: 'no',
+    ifYesIndigenousCattle: 123132123,
+    sheep: 'no',
+    ifYesSheep: 12312312,
+    goats: 'no',
+    ifYesGoats: 312123,
+    camels: 'no',
+    ifYesCamels: 312123,
+    donkeys: 'no',
+    ifYesDonkeys: 213312,
+    ifYesChicken: 2,
+    howManyBirths: 0,
+    howManyDeaths: 0,
+    householdConditions: 'poor',
+    skipMeals: 'no',
+    receivingBenefits: 0,
+    ifYesNameProgramme: 0,
+    typeOfBenefit: 'in_kind',
+    ifOtherBenefit: 2123312,
+    ifCash: 12312,
+    ifInKind: 132132,
+    feedbackOnRespons: 'no',
+    ifYesFeedback: 312123,
+    whoDecidesHowToSpend: 'male_household_head',
+    possibilityForConflicts: 'no',
+    genderedDivision: 'no',
+    ifYesElaborate: 'asddas',
+    geopoint: 123231,
+  };
+
+  describe('with FSP: Safaricom', () => {
+    let accessToken: string;
+
+    beforeEach(async () => {
+      await resetDB(SeedScript.krcsMultiple);
+      accessToken = await getAccessToken();
+    });
+
+    it('should successfully pay-out', async () => {
+      // Arrange
+      await importRegistrations(
+        programId,
+        [registrationSafaricom],
+        accessToken,
+      );
+
+      await awaitChangePaStatus(
+        programId,
+        [registrationSafaricom.referenceId],
+        RegistrationStatusEnum.included,
+        accessToken,
+      );
+      const paymentReferenceIds = [registrationSafaricom.referenceId];
+
+      // Act
+      const doPaymentResponse = await doPayment(
+        programId,
+        payment,
+        amount,
+        paymentReferenceIds,
+        accessToken,
+      );
+
+      await waitForPaymentTransactionsToComplete(
+        programId,
+        paymentReferenceIds,
+        accessToken,
+        3001,
+        Object.values(StatusEnum),
```

Comment:
I think you should not include status 'waiting' here. Else the test could break if you get the transactions before the callback from the mock-service happens and that transaction is still on 'waiting'

Reply:
done